Updates for Kim (#56)
mulesoft/docs-partner-manager: modules/ROOT/pages/configure-message-flows.adoc

> endif::[]

Reviewer: HIDDEN

@@ -1,12 +1,12 @@
= Create a Message Flow
ifndef::env-site,env-github[]
include::_attributes.adoc[]

> To receive messages from a trading partner, create a message flow and configure the sections for the source, map, and target.

Reviewer: ```suggestion
Before your organization as host can receive messages from your customer trading partner, you must create a _message flow_ and configure the sections within that flow for the _source_, _map_, and _target_.
```

Reviewer: ADDENDUM
I have another idea. Why not make this topic into three tasks:

Create a Message Flow
   Create the Source: definition, requirements, steps
   Create the Map: ditto
   Create the Target: ditto

Author: Will look at next release.

> Your organization is typically referred to as the host. The enterprise or customer you interact with is referred to as your partner.

Reviewer: OMIT per inclusion above. 
This isolated sentence of definitions sounds awkward, and the definitions have already been established in previous texts.

Author: PM wants it here. I will have to negotiate with him.

> The source of the message flow is defined by the partner, a receiving endpoint at the host (often an API) and the message type you are receiving from the partner.

Reviewer: ```suggestion
The source of the message flow is defined by the partner, a receiving endpoint at the host (often an API), and the message type you are receiving from the partner.
```
Does this mean that the message flow is determined by three separate pieces of info: who the partner is, where the receiving host endpoint is, and what type the message is? I'm confused, because the target is the receiving endpoint. I think there must be a simpler way to describe this, without overlap.

That's why I wonder about making this three separate task topics....

> The target is the final destination of a message flow and is defined by a host message type and endpoint, often a Host API.

Reviewer: ```suggestion
The target is the final destination of a message flow and is defined by a host message type and endpoint, often a host API.
```
Is the host API both the type and endpoint, or just the endpoint?

> 

Reviewer: HIDDEN
Edited; review for changes

To configure the source portion of your message flow, you need the following information:

* Your partner's name and associated identifiers (such as X12-ISA, X12 GS, and so on)
* The endpoint where you (host) receive transmissions
* The message type to process for this message flow (such as X12, 4010, 850, and so on).
* The endpoint where your partner might receive an acknowledgment, if required
* The DataWeave map that transforms the received message contents into a format you can use

To configure the target portion of the message flow, you need the following information:

* A message type that your system understands
This is also the format into which the DataWeave map transforms received messages.
* The internal host endpoint to which you route the transformed message

If a separate URL is required for returning MDNs, it is sent in the AS2 transmission and Anypoint Partner Manager reads it from there. 
//To what part of the previous text does this apply? Source? Map? Target? Something different?//

== Configure a Message Flow
//This is an odd heading hierarchy, to call the topic Create a Message Flow and follow it with a LONG topic intro that defines the three parts, and then have only one subtopic that actually creates the flow but is called Configure a Message Flow. See suggestion for breaking into three tasks.//

. From within your sandbox or production Anypoint Partner Manager environment, click *New message flows* in the upper right.
. In the *Select a partner* dialog, search for and select a partner name, or click *New* to create a new partner name.

Author: Needs work on the next release.

. Once you find a match, choose the partner name and click *Select*.

* Alternatively from the *Select a partner* dialog, you can create a new partner by clicking *New*.
This is the same as xref:configure-partner.adoc#create-a-trading-partner[Create a Trading Partner] except that when you are finished, you'll click *Create Partner and Message Flow*.

> You can add more information to the partner later.

Reviewer: OMIT all of this by rewording Step 2...which matches Step 4 below.

+

> NOTE: Clicking each section of the message flow expands it. A green checkmark next to any a section indicates that it is complete.

Reviewer: Why is this note here? It is not related to either the step above or below it, and it applies to the whole process, not this part of the process. 
This would be better before the first step, if documented at all....

Author: I know. The PM insisted that it go here. I have tried to take it out. I just moved it.

> NOTE: Clicking each section of the message flow expands it. A green checkmark next to any a section indicates that it is complete.

Reviewer: HIDDEN

. In the Receiving endpoint section, click *Select* to configure the source endpoint (where the host receives the transmission).
. Click *Select* to search for and select an existing endpoint, or click *New* to create a new endpoint.

//Does the user click *Select* twice; is this correct, or is the second of these two steps repeating that action erroneously?//

. To create a new endpoint, depending on the protocol, refer to the following:
+
* xref:endpoint-as2-receive.adoc[AS2 Endpoint Settings]

> * xref:endpoint-sftp-receive-target.adoc[SFTP Receive Endpoint Settings]

Reviewer: Suggest removing these cross-refs., unless your audience really has to use them to follow up on clicking *New*....

Author: Yes, they do need these.

+

> . Click *Select* from the *Source message* section to expand the Message Type section.

Reviewer: ```suggestion
. In the *Source message* section, click *Select* to expand the Message Type section.
```

> . Click *Select* from the *Source message* section to expand the Message Type section.

Reviewer: HIDDEN
Edited; please compare

. In the Message Type section, either search for and click *Select* to choose a previously used message type, or click *New* to create a new message type.

. For a new message type, depending on the message standard, refer to the following:
+

> * xref:document-types.adoc[Message Types]

Reviewer: ditto re. cross-refs interrupting task step flow

Author: Yes that's true but there are several procedures for certain items that are repeated over and again in different parts of the docs. I need to go over this with you for the next release and come up with a better solution.

+

> . (Optional) Click the *View* link to edit the validation settings and then click the hostname.

Reviewer: ```suggestion
. To optionally edit the validation settings, click *View* and the hostname.
```

> For more information, see xref:x12-receive-read-settings.adoc[X12 Settings - Receive from Partner].

Reviewer: ditto re. cross-refs

Author: Let's look at this next release.

> . (Optional for X12 partner requirements) Configure the type of functional acknowledgment, and the endpoint to which you send it. Refer to the following for endpoint information:

Reviewer: ```suggestion
. For X12 partner requirements, optionally configure the type of functional acknowledgment and the endpoint to which you send it. 
```
ditto re. cross refs

+
* xref:endpoint-https-send.adoc[HTTP and HTTPS Send Configuration]

* xref:endpoint-sftp-receive-target.adoc[SFTP Receive Endpoint Settings]

> +

Reviewer: ditto

> . Click the *Map* section and then click *Import* to choose a DataWeave map.

Reviewer: This could represent the beginning of a Create a Map task within Create a Message Flow.
You could add a subhead, and an intro something like...:

"After you finish creating the source of your message flow, you continue working in the *New message flows* link within your sandbox or production Anypoint Partner Manager environment to create the map portion of your message flow."

Author: Yes, let's look at that for next release.

> 

Reviewer: HIDDEN
Edited; please compare

. Select a map and click *Upload mapping*.

//This could be the beginning of the Create a Target task, with the same type of text intro as for the other two tasks. I see that this is short, as was the Create a Map task, so...I guess I know why they are currently all one task. //

. Click the *Target at <Partner name>* section and click *Select* to set up a target message type and endpoint to which you want to route the transformed message.

> . Either search and select an existing message type, or create a new message type. To create a new message type, depending on the message standard, follow instructions in one of these sections.

Reviewer: ```suggestion
. Either search and select an existing message type, or create a new message type.
To create a new message type, depending on the message standard, follow instructions in one of these sections:
```
But suggest avoiding xrefs in steps. Can we move all these to See Also?

Author: Hmm, that's one way - I'd like to explore other ways organizationally.

+
* xref:document-types.adoc[Message Types]
+

> .  Click *Select* to set up a target endpoint. Either search and select and existing endpoint, or create a new endpoint.

Reviewer: Is this one step or two? Does the user click *Select* once or twice?
From what location does the user click *Select* in the first sentence?
```suggestion
.  Click *Select* to set up a target endpoint. 
. Either search for and select an existing endpoint, or create a new endpoint by clicking *New*.
```
?

+
* To create a new endpoint, depending on the protocol, refer to the following:

* xref:endpoint-sftp-receive-target.adoc[SFTP Receive Endpoint Settings]

* xref:endpoint-https-send.adoc[HTTP and HTTPS Send Configuration]

> 

Reviewer: ditto

Reviewer: What follows could be a new topic: Deploy Your Message Flow.
If so, we might want to change the container topic to Create and Deploy a Message Flow.

Author: It will actually be - create, deploy, and test a message flow next release.

+

> When you complete all of the sections (they show a green check mark next to all of them in the flow) you can deploy your message flow and start receiving messages from your partner.

Reviewer: ```suggestion
After you create the source, map, and target sections in the interface, confirm that each shows a green check mark next to it in the flow), and click *Deploy*.

Your message flow configuration is deployed to your Mule runtime engine and you can start receiving messages from your partner. Use the Activity page to track processed messages.
```

Reviewer: HIDDEN
Edited: delete everything but ==See Also

. Click Deploy when you are ready to deploy the configurations to your Mule Runtime.
Once a message flow is successfully deployed, you can track its processed messages from the Activity page.
== See Also

Author: For next release.
